cpp/src/join/mixed_join_kernels_semi.cuh

```diff
@@ -51,16 +51,17 @@ namespace detail {
  * @param[in] device_expression_data Container of device data required to evaluate the desired
  * expression.
  */
-template <cudf::size_type block_size, bool has_nulls>
-__global__ void mixed_join_semi(table_device_view left_table,
-                                table_device_view right_table,
-                                table_device_view probe,
-                                table_device_view build,
-                                row_hash const hash_probe,
-                                row_equality const equality_probe,
-                                cudf::detail::semi_map_type::device_view hash_table_view,
-                                cudf::device_span<bool> left_table_keep_mask,
-                                cudf::ast::detail::expression_device_view device_expression_data);
+template <cudf::size_type block_size, bool has_nulls, typename HashProbe>
+__attribute__((visibility("hidden"))) __launch_bounds__(block_size) __global__
```

Comment:
We have done some refactoring of our kernel visibility recently. I don't know what set of modifiers this needs for its visibility, but I think it might be the same as this one? cc: @robertmaynard to check this.

https://github.com/rapidsai/cudf/blob/f5113228c3aa89d49e71d42d11c38afe52695aa6/cpp/src/join/mixed_join_kernel.cuh#L43

```suggestion
CUDF_KERNEL __launch_bounds__(block_size)
```

Reply:
Yes, this PR needs to be refactored to follow the new layout in https://github.com/rapidsai/cudf/pull/16603

Reply:
fixed